rainbow-cake-core/src/main/java/co/zsmb/rainbowcake/base/RainbowCakeDialogFragment.kt

```diff
@@ -0,0 +1,96 @@
+package co.zsmb.rainbowcake.base
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.CallSuper
+import androidx.annotation.LayoutRes
+import androidx.fragment.app.DialogFragment
+import co.zsmb.rainbowcake.internal.logging.log
+
+/**
+ * Base class for DialogFragment that connects them to the appropriate ViewModel instances.
+ */
+public abstract class RainbowCakeDialogFragment<VS : Any, VM : RainbowCakeViewModel<VS>>
+    : DialogFragment() {
+
+    private val logTag: String by lazy(mode = LazyThreadSafetyMode.NONE) { "RainbowCakeDialogFragment ($this)" }
+
+    /**
+     * The ViewModel of this Fragment.
+     */
+    protected lateinit var viewModel: VM
+
+    @CallSuper
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        viewModel = provideViewModel()
+
+        viewModel.events.observe(this) { event ->
+            event?.let { onEvent(it) }
+        }
+        viewModel.queuedEvents.observe(this) { event ->
+            event?.let { onEvent(it) }
+        }
+    }
+
+    /**
+     * Calls to super for this method MAY be omitted if the View inflation needs
+     * to be customized. In these cases, [getViewResource] can return any value
+     * as it won't be used (recommendation: 0).
+     */
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val viewResource = getViewResource()
+        require(viewResource != 0) {
+            "Override getViewResource to provide a layout resource ID, or override onCreateView to provide a layout directly"
+        }
+        return inflater.inflate(viewResource, container, false)
+    }
+
+    @CallSuper
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel.state.observe(viewLifecycleOwner) { viewState ->
+            viewState?.let { render(it) }
+        }
+    }
+
+    /**
+     * This method should return a ViewModel instance for the current Fragment.
+     *
+     * If one of RainbowCake's own DI libraries are being used, this method should
```

Comment:
Isn't _If one of RainbowCake's own DI libraries **is** being used_ the grammatically correct choice of word here?